### ISSUE TITLE
Looking for Maintainers!

### ISSUE THREAD

**Issue by D3DeFi** [ansible-collections/community.zabbix#466]
##### Help wanted
Hello Ansible Zabbix community!

We are looking for maintainers, reviewers, contributors and everyone else that may help with moving this collection forward.

Unfortunately, in the last months, this collection has become more stale and maintainer activity ceased down to the minimum.

Please comment on this issue or contact me directly at d3defi@gmail.com if you are interested.

**Comment by D3DeFi** [comment]
@gundalow maybe this could also be highlighted in the bullhorn?

I don't plan to step down, but this collection mostly come to halt as work on the plugins and roles seem to be around 10% compared to what it used to be when we first created the collection.

It has, unfortunately, became increasingly difficult for me to find spare time and energy.

**Comment by miettal** [comment]
I have little experience in English skills and developing OSS, but I would like to cooperate as much as I can.
I think a simple issue can be fixed.

**Comment by stefanux** [comment]
happy to help, will contact you directly.
If someone knows any sponsors who are willing to put a couple of bucks into this: this might help too!

The role (and now the collection) already helped countless Admins/Devops/SREs all over the world, would hate to see it dying a slow death.

**Comment by D3DeFi** [comment]
Wow, I have expected this post to go unnoticed for some time :sweat_smile: Thank you all for offering your help! I really appreciate it :)

It goes without saying that we need bugs fixed and features implemented (especially closing issues that are open). There is also space for new modules to be included as we do not cover all of the options that API offers (trends, web scenarios, etc..).

Here are things that come to my mind we need help with:
* PR reviews & approvals (validating the PRs and ensuring they include tests & changelog fragments)
* improving our CI (more tests, refactoring/simplifying/optimizing what we have already, improving how can we test locally & maybe multiple versions in parallel)
* dropping Zabbix 3.X support in modules + cleaning up a code base related to 3.X hacks
* enabling plugins CI against Zabbix 5.4 and fixing failing test suites
* starting preparations for Zabbix 6.0 both for roles and modules
* improving on repository management (automatic labeling of issues/PRs, /rebase option for contributors if possible, any other ideas/bots that may help)
* improving documentation (modules, roles, repo)
* there is also an effort to migrate part of the shared modules functionality to module_utils (base classes, frequent zbx.XYZ.get() calls, e.g. identifying parts of code that can be reused across all modules)

Lets try it like this first - feel free to comment under issues that you would like to work on them. I will then assign them to you so we can see who works on what. If you want to work on something new, either open new issue, comment here or bring it up on Gitter so we don't duplicate effort. Once you get a hold of "life" in this community and would still like to get more responsibilities we can promote you to actual repository maintainers :)

I am also open to suggestions if you think there is more optimal way to approach this "on-boarding" process.

Also please try to find a few moments to understand our guidelines:
* https://github.com/ansible-collections/community.zabbix#supported-zabbix-versions
* https://github.com/ansible-collections/community.zabbix/blob/main/CONTRIBUTING.md
* https://github.com/ansible-collections/community.zabbix/blob/main/docs/RELEASE.md

Also you are welcome to log into our Gitter and chat freely. I try to respond there when tagged, but usually not immediately:
https://gitter.im/community-zabbix/community

Btw I am mostly speaking from a perspective of modules maintainer. @dj-wasabi is there something specific you would need a help with regarding roles?

**Comment by dj-wasabi** [comment]
Well, if someone has a Windows node available that might help validating the Agent on Windows issues/PR's. Other than that, any help is welcome. 👍

**Comment by D3DeFi** [comment]
Hello everyone!

Sorry for my very delayed response. I am spread too thinly these months and only have time to react to the most pressing issues in the community.

@miettal @stefanux I've seen you guys lurking around the repository a bit commenting on PRs/opening some bugfixes. Since you already had a chance to "taste a maintainership" in this repository, what are your opinions? Would you like to further continue?

As I previously said, we urgently need help with management around the repository (reviews, preparing releases, directing repository, managing "project" like zbx6.0 support), but also a little help with actual coding. 

@lzadjsf would you be interested in lurking around the repository helping with issues/PRs as I requested from other contributors to see if you would like to do such activities?

I hope you all understand reasons for not showering you with maintainer permissions right away :) Thank you all for the patience and your interest <3

**Comment by lzadjsf** [comment]
Huh... Seems it's time to make my contribution into development zabbix and ansible roles for it.
Yes, let's try.

**Comment by stefanux** [comment]
sure, not familiar with molecule atm, but happy to help from sysadmin / hosting stuff / testing-angle. i believe @dj-wasabi mentioned a windows testing-VM?

**Comment by dj-wasabi** [comment]
@stefanux  Yes, we need to do something with Windows as discussed in #540 . It seems that quite a lot of people are using it, so we just need to find a way to test the Agent on Windows so we don't break things. If you do have any experience with testing it on Windows, please let me know. I think we should focus on a single Windows version for now.

But as we have some new maintainers added, I will close this issue.

But if you still want to help, please do so! Any help is really appreciated!

**Comment by D3DeFi** [comment]
Here we go again :)

**Comment by lzadjsf** [comment]
For sure!

**Comment by BGmot** [comment]
Looking at previous comments in this issue - do we still have problems with modules regarding 6.0? I committed some code to fix all issues known to me and now we have all the tests running against Zabbix 6.0, no failures.

**Comment by D3DeFi** [comment]
> Looking at previous comments in this issue - do we still have problems with modules regarding 6.0? I committed some code to fix all issues known to me and now we have all the tests running against Zabbix 6.0, no failures.

Sorry, we dont have any major problems with 6.0 I am aware of. You did a great job covering requirements for 6.0 lately.

I updated the issue description. Btw would you also consider stepping up to the maintainer status?

**Comment by BGmot** [comment]
No need to apologize -) I just was worried that I missed some incompatibilities with 6.0

> Btw would you also consider stepping up to the maintainer status?

I guess I would as I am really interested in keeping this developing... but I never maintained anything and many aspects still mystery for me (like when you need to provide a changelog update or not, for instance). I am not a developer.

**Comment by pyrodie18** [comment]
Would love to spend time helping out in whatever capacity you need.  I maintain a number of things for work (not public, unfortunately) but am open to whatever you need.

**Comment by D3DeFi** [comment]
@gundalow are you around? I tried to contact you via Gitter.

Can you please help me with including new maintainers to the collection?

**Comment by Andersson007** [comment]
@D3DeFi hi, sorry for the late reply, was on PTO which started exactly 14 days ago when you wrote the comment:)
I invited @pyrodie18 with "write" access. @pyrodie18 could you please check your mailbox? And thank you for your interest and help!
@D3DeFi do we wanna make anyone else maintainers? Next time ping me, not gundalow, I respond and am not gonna have another PTO soon. It's a great idea to involve other folks in maintenance!

**Comment by D3DeFi** [comment]
@Andersson007 thank you!

Can you also include @BGmot and @lzadjsf ?

**Comment by Andersson007** [comment]
@D3DeFi invited the folks too

> Is there also a way I can ask you direct questions somehwere without polluting comments section?

yes, sure, aklychko at redhat dot com or andersson007_ on [Matrix](https://docs.ansible.com/ansible/devel/community/communication.html#ansible-community-on-matrix)

**Comment by Andersson007** [comment]
BTW we have the [maintainer guidelines](https://docs.ansible.com/ansible/latest/community/maintainers.html). Maybe it doesn't fully match your practices in this collection but might be useful to read (especially for new mainteiners).

**Comment by D3DeFi** [comment]
Yes, thanks for linking maintainer guidelines. We don't have any specific rules in the community, so everything that is required by upstream goes. Anyway, I am always open to any suggestions if you have any ideas how to improve our workflows in here.

If you guys start reading from https://github.com/ansible-collections/community.zabbix/blob/main/README.md#supported-zabbix-versions and follow the links, you should get a good feel about how to "respond" to daily situations in the repository.

I especially encourage you to:
- review PRs (means write comments and hit 'approve/request changes/comment' toggles on the reviews)
- comment on issues & help reporters to resolve their problems
- assign issues to yourself if you plan to fix them
- label PRs and issues

As for merging stuff, unspoken rule is that we do `squash and merge` and we want to have at least 2 maintainers reviewing the code before merging. (Ignore last weeks where I reviewed and merged stuff alone as I had no-one else to request reviews from).

If the PR contains small typo fixes or non-invasive changes, feel free to merge right away if you are sure that it doesn't break anything.

Discussing stuff (personally I try to respond asap, but have also other priorities so may get delayed by a couple of days sometimes):
- open issues or discussions in this repo and link people you would like to respond
- brainstorm/chat here: https://gitter.im/community-zabbix/community
- AMA at d3defi@gmail.com 

Releasing new versions were always done by me so far. Next time I will go through 1.8.0 release I will note what is not documented in our docs/ and fill in the gaps so it is not dependent on me. ( side note: For one I know I tend to add missing changelog bits by hand before tagging a commit as a release - would be awesome if we can force users to add changelogs via CI ).

Looking forward to working with you all! :rocket: :heart:

**Comment by D3DeFi** [comment]
> @D3DeFi invited the folks too
> 
> > Is there also a way I can ask you direct questions somehwere without polluting comments section?
> 
> yes, sure, aklychko at redhat dot com or andersson007_ on [Matrix](https://docs.ansible.com/ansible/devel/community/communication.html#ansible-community-on-matrix)

@Andersson007 thanks for providing official channels to you. I will contact you if something arises

**Comment by D3DeFi** [comment]
Also closing this issue for the time being :) Feel free to still comment here if you want to add something

**Comment by D3DeFi** [comment]
@mu1f407 you have been such a great help lately around the repo. If you think you would be also interested in joining us as a maintainer, feel free to hit me up anytime with any questions :)

**Comment by mu1f407** [comment]
@D3DeFi Yeah, I've been thinking about it for a while. I saw some of original maintainers (dj-wasabi, rockaut) had to step out due to lack of time or change of interest. As currently we use modules from this collection in our company and I hope I will have a little bit of time in foreseeable future, I'll be glad to help. Thanks. 🙂

**Comment by D3DeFi** [comment]
@Andersson007 can you please also invite @mu1f407 ? :)

**Comment by Andersson007** [comment]
@D3DeFi done, @mu1f407 could you please check your email for the invite? Thank you for your interest and help!

Please take a look at:
- the comment by @D3DeFi , well-written, I would also add (fixing CI and releasing as it would be nice if someone else can do it too)
- the [Maintainer guidelines](https://docs.ansible.com/ansible/latest/community/maintainers.html)
Thanks

**Comment by Andersson007** [comment]
@mu1f407  @BGmot @lzadjsf please subscribe and track the [news-for-maintainers](https://github.com/ansible-collections/news-for-maintainers) repo as important CI update announcements that we must do happen there

**Comment by mu1f407** [comment]
@Andersson007 I'm in, thank you! Subscribed too. I will read through guidelines, news and so on in the next days.
@D3DeFi Great introducing post. Agree with everything. Nice to see maintainer community expanding.

**Comment by pyrodie18** [comment]
@Andersson007 can I get a list of who all has write access to this repo?  I'm tracking myself and @BGmot.  Pretty sure @D3DeFi still does too.  Who else?

Thanks,

Troy

**Comment by dj-wasabi** [comment]
I am.

**Comment by Andersson007** [comment]
@pyrodie18 sure

besides you: BGmot, D3DeFi, dj-wasabi, ragingpastry, rockaut, sky-joker
if somebody hasn't participated for a long time/not interested explicitly and needs to be removed, let me know

**Comment by pyrodie18** [comment]
> I am.

@dj-wasabi  are you still at all active?  Do you want to stay on as a maintainer?

How about you guys @D3DeFi  and @ragingpastry, @rockaut , and @sky-joker ?

**Comment by BGmot** [comment]
I know that I am not very active lately but I definitely hope to get back and keep contributing/maintaining.

**Comment by D3DeFi** [comment]
I am so sorry, but I don't see myself being able to actively participate, so you can go ahead and remove my access. If you need some insights in the future, feel free to tag me in the issues however

**Comment by Andersson007** [comment]
> I am so sorry, but I don't see myself being able to actively participate, so you can go ahead and remove my access. If you need some insights in the future, feel free to tag me in the issues however

@D3DeFi thanks for the feedback! Removed, though I'll be happy to elevate the permissions again once you feel able again, just let us know!
Anyway, thanks for your great long term contribution and the help offer!

**Comment by dj-wasabi** [comment]
> > I am.
> 
> @dj-wasabi are you still at all active? Do you want to stay on as a maintainer?
> 
> How about you guys @D3DeFi and @ragingpastry, @rockaut , and @sky-joker ?

I don't have any time, so maybe it is better for removing my maintainer access.

Thank you all for doing a good job though! And please take good care of my Ansible roles please 😉

**Comment by pyrodie18** [comment]
Thank you for the time and effort @dj-wasabi  and @D3DeFi

**Comment by dj-wasabi** [comment]
Any time! And dont hesitate to reach out when you need opinions, suggestions or anything. Where needed - and have the time - i will happily help out!

**Comment by Andersson007** [comment]
@dj-wasabi thank you much for your contributions and being in touch! You'll get back as a maintainer as soon as you let us know that you want to

**Comment by pyrodie18** [comment]
@Andersson007 judging by the fact that they haven't replied back, and just looking at their general level of activity on Github, can you please remove @ragingpastry , @rockaut , and @sky-joker .  Also can you please add @eb4x and @masa-orca .  Welcome to the team guys.  There's a few links above that you should check out.

**Comment by Andersson007** [comment]
@pyrodie18 done, thanks! And thanks for promoting the folks to new maintainers - many maintainers forget about it but it's very important!

@ragingpastry @rockaut @sky-joker thank you for your great and long-term contributions! Once you're back, let us know, you'll be, of course, welcome as maintainers again!

@eb4x @masa-orca thanks for joining the crew! Just a fyi, we have the [Collection maintainer guidelines](https://docs.ansible.com/ansible/devel/community/maintainers.html#maintainers) that highlight important things and can be generally and practically helpful. Thanks!

**Comment by dj-wasabi** [comment]
> @pyrodie18 done, thanks! And thanks for promoting the folks to new maintainers - many maintainers forget about it but it's very important!
> 
> @ragingpastry @rockaut @sky-joker thank you for your great and long-term contributions! Once you're back, let us know, you'll be, of course, welcome as maintainers again!
> 
> @eb4x @masa-orca thanks for joining the crew! Just a fyi, we have the [Collection maintainer guidelines](https://docs.ansible.com/ansible/devel/community/maintainers.html#maintainers) that highlight important things and can be generally and practically helpful. Thanks!

Congrats @eb4x  & @masa-orca ! 🥳

**Comment by eb4x** [comment]
> Congrats @eb4x & @masa-orca ! 🥳

:bow:
